src/main/java/com/project/videotecha/config/SecurityConfiguration.java

```diff
@@ -0,0 +1,92 @@
+
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.jwk.source.ImmutableJWKSet;
+import com.nimbusds.jose.jwk.source.JWKSource;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.project.videotecha.service.UserService;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.annotation.web.configurers.oauth2.server.resource.OAuth2ResourceServerConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtEncoder;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+import org.springframework.security.oauth2.jwt.NimbusJwtEncoder;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationEntryPoint;
+import org.springframework.security.oauth2.server.resource.web.access.BearerTokenAccessDeniedHandler;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.web.cors.CorsConfiguration;
+import org.springframework.web.cors.CorsConfigurationSource;
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
+
+import java.util.List;
+
+@Configuration
+@EnableWebSecurity
+@EnableMethodSecurity
+public class SecurityConfiguration {
+
+    private final RsaKeyProperties jwtConfigProperties;
+    private final UserService userService;
+
+    public SecurityConfiguration(RsaKeyProperties jwtConfigProperties, UserService userService) {
+        this.jwtConfigProperties = jwtConfigProperties;
+        this.userService = userService;
+    }
+
+    @Bean
+    public InMemoryUserDetailsManager user() {
+        com.project.videotecha.model.User user = userService.getById(1L);
+        return new InMemoryUserDetailsManager(User
+                .withUsername(user.getEmail())
+                .password(user.getPassword())
+                .authorities(user.getUserType().name()).build());
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(auth -> auth.requestMatchers("/auth/*").permitAll().anyRequest().authenticated())
```

Comment:
Consider using /** instead of /* while matching auth paths. The latter matches any value after /, while the former catches any value after / including another /.

Currently this is fine and covers all the endpoints which are needed but it could present an error down the line. Currently this covers paths like `/auth/endpoint1` or `/auth/endpoint2` but doesn't work for `/auth/endpoint1/endpoint2`